src/docfx/build/redirection/RedirectionMap.cs

```diff
@@ -47,7 +47,7 @@ public static RedirectionMap Create(ErrorLog errorLog, Docset docset, Func<strin
             AddRedirections(docset.Config.Redirections, redirectDocumentId: true);
             var redirectionsByRedirectionUrl = redirections
                 .GroupBy(file => file.RedirectionUrl, PathUtility.PathComparer)
-                .ToDictionary(group => group.Key, group => group.First(), PathUtility.PathComparer);
+                .ToDictionary(group => group.Key, group => group.OrderBy(r => r).First(), PathUtility.PathComparer);
```

Comment:
If we don't order here, we pick the first one that appears in redirection.json, the implication is that writers can safely add new redirection rules to the end and does not affect document_id of existing articles.

If we sort here, the above assumption is broken, new redirection rules may change document_id of an existing article. What is the expected behavior here?

Reply:
U r right, we should pick up the first on wrote by user